hrqb/base/task.py

```diff
@@ -1,17 +1,20 @@
 """hrqb.base.task"""
 
+import logging
 import os
 from abc import abstractmethod
+from collections.abc import Iterator
 from typing import Literal
 
 import luigi  # type: ignore[import-untyped]
 import pandas as pd
 
 from hrqb.base import PandasPickleTarget, QuickbaseTableTarget
```

Comment:
Can the `target` attribute be converted to a property on these `HRQBTask` classes using `@property`? 🤔

Reply:
Am I understanding that you'd want to be able to use `self.target` instead of `self.target()` in places like this?

```python
class PandasPickleTask(HRQBTask):    

    def run(self) -> None:
        self.target.write(self.get_dataframe())   # <----------------?
```

If so, I think it'd get tricky.  `target()` is a luigi method from `luigi.Task` that `HRQBTask` extends.  Actually, one reason these classes exist is so that actual downstream tasks won't have to interact with `self.target()` very often.  They should like either define `get_dataframe()` or `get_records()`, which themselves interact with `self.target()`. 

In short: `target` is a reserved method from `luigi.Task`, so could be hardy to property-ify.

Reply:
Thanks for the conversation on slack.  During that, realizing that I misspoke above and `target` is _not_ a reserved property or method on `luigi.Task`.  As discussed, I think this was me early on in the project thinking it'd be handy to refer directly to a Task's "target" as an object to receive and work with vs what feels like a more opaque `output()` method which just so happens to return a target class.

Going to dig in a bit and see if some renaming or reworking might make this a little cleaner.  Thanks for raising!

Reply:
@jonavellecuerdo - this commit was added that addresses your original request / question: https://github.com/MITLibraries/hrqb-client/pull/14/commits/142cfb80f0727d8cf5fe96ec5e54809bcd492f4a.

Overall, I think it's a good one, thanks for raising.

To summarize the commit message a bit: the goal of `HRQBTask.target()` was to more accurately reflect that it's returning an actual `luigi.Target` instance, where the default requires in luigi are to define a `<Task>.output()` method that does the same.  All in all, it's mostly just a project-specific preference to use the word "target" (which feels more accurate to what it returns) over the default luigi "output".  But by making it a property, as you asked about, I think it further leans into this opinionation in a good way!  

The end result is the ability to do `MyTask.target` and get the fully instantiated target instead of `MyTask.output()` which, to me, feels a bit vague as to what it's returning or doing.